src/main/java/org/gaul/s3proxy/Main.java

```diff
@@ -117,14 +121,42 @@ public static void main(String[] args) throws Exception {
 
             String s3ProxyAuthorizationString = properties.getProperty(
                     S3ProxyConstants.PROPERTY_AUTHORIZATION);
+            ImmutableList.Builder<String> locatorBuckets =
+                    new ImmutableList.Builder<>();
+            for (String key : properties.stringPropertyNames()) {
+                if (key.startsWith(S3ProxyConstants.PROPERTY_BUCKET_LOCATOR)) {
+                    locatorBuckets.add(properties.getProperty(key));
+                }
+            }
+            ImmutableList<String> buckets = locatorBuckets.build();
+            String localIdentity = null;
+            String localCredential = null;
             if (AuthenticationType.fromString(s3ProxyAuthorizationString) !=
                     AuthenticationType.NONE) {
-                String localIdentity = properties.getProperty(
+                localIdentity = properties.getProperty(
                         S3ProxyConstants.PROPERTY_IDENTITY);
-                String localCredential = properties.getProperty(
+                localCredential = properties.getProperty(
                         S3ProxyConstants.PROPERTY_CREDENTIAL);
-                locators.put(localIdentity, Maps.immutableEntry(
-                        localCredential, blobStore));
+                Map.Entry<String, String> key = Maps.immutableEntry(
+                        localIdentity, null);
+                if (parsedLocations.add(key)) {
+                    locators.put(key, Maps.immutableEntry(
+                            localCredential, blobStore));
+                }
+            }
+            if (!buckets.isEmpty()) {
```

Comment:
Do you need to check whether buckets is empty?  The foreach loop should handle this already.